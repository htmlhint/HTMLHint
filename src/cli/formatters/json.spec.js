const run = require('../../../test/test-utils').run
const path = require('path')
const serializer = require('jest-serializer-path')

expect.addSnapshotSerializer(serializer)

describe('CLI', () => {
  describe('Formatter: json', () => {
    it('should have stdout output with formatter json', async () => {
      const { exitCode, stdout, stderr } = await run(__dirname, [
        path.resolve(__dirname, '__fixtures__', 'example.html'),
        '--format',
        'json',
      ])

      expect(exitCode).toBe(1)

      expect(
        JSON.parse(stdout.replace(/"time":\d+/, '"time":99'))
      ).toMatchSnapshot()

      expect(stderr).toBe('')
    })
  })
})
