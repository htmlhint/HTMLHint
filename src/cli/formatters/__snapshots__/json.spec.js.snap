// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI Formatter: json should have stdout output with formatter json: stderr 1`] = `""`;

exports[`CLI Formatter: json should have stdout output with formatter json: stdout 1`] = `
Array [
  Object {
    "file": "<PROJECT_ROOT>/src/cli/formatters/__fixtures__/example.html",
    "messages": Array [
      Object {
        "col": 7,
        "evidence": "  <div bad='' bad=''></div>",
        "line": 8,
        "message": "The value of attribute [ bad ] must be in double quotes.",
        "raw": " bad=''",
        "rule": Object {
          "description": "Attribute values must be in double quotes.",
          "id": "attr-value-double-quotes",
          "link": "https://github.com/thedaviddias/HTMLHint/wiki/attr-value-double-quotes",
        },
        "type": "error",
      },
      Object {
        "col": 14,
        "evidence": "  <div bad='' bad=''></div>",
        "line": 8,
        "message": "The value of attribute [ bad ] must be in double quotes.",
        "raw": " bad=''",
        "rule": Object {
          "description": "Attribute values must be in double quotes.",
          "id": "attr-value-double-quotes",
          "link": "https://github.com/thedaviddias/HTMLHint/wiki/attr-value-double-quotes",
        },
        "type": "error",
      },
      Object {
        "col": 14,
        "evidence": "  <div bad='' bad=''></div>",
        "line": 8,
        "message": "Duplicate of attribute name [ bad ] was found.",
        "raw": " bad=''",
        "rule": Object {
          "description": "Elements cannot have duplicate attributes.",
          "id": "attr-no-duplication",
          "link": "https://github.com/thedaviddias/HTMLHint/wiki/attr-no-duplication",
        },
        "type": "error",
      },
      Object {
        "col": 7,
        "evidence": "  <div bad='' bad=''></div>",
        "line": 9,
        "message": "The value of attribute [ bad ] must be in double quotes.",
        "raw": " bad=''",
        "rule": Object {
          "description": "Attribute values must be in double quotes.",
          "id": "attr-value-double-quotes",
          "link": "https://github.com/thedaviddias/HTMLHint/wiki/attr-value-double-quotes",
        },
        "type": "error",
      },
      Object {
        "col": 14,
        "evidence": "  <div bad='' bad=''></div>",
        "line": 9,
        "message": "The value of attribute [ bad ] must be in double quotes.",
        "raw": " bad=''",
        "rule": Object {
          "description": "Attribute values must be in double quotes.",
          "id": "attr-value-double-quotes",
          "link": "https://github.com/thedaviddias/HTMLHint/wiki/attr-value-double-quotes",
        },
        "type": "error",
      },
      Object {
        "col": 14,
        "evidence": "  <div bad='' bad=''></div>",
        "line": 9,
        "message": "Duplicate of attribute name [ bad ] was found.",
        "raw": " bad=''",
        "rule": Object {
          "description": "Elements cannot have duplicate attributes.",
          "id": "attr-no-duplication",
          "link": "https://github.com/thedaviddias/HTMLHint/wiki/attr-no-duplication",
        },
        "type": "error",
      },
      Object {
        "col": 22,
        "evidence": "  <input type=\\"text\\"></input>",
        "line": 10,
        "message": "Tag must be paired, no start tag: [ </input> ]",
        "raw": "</input>",
        "rule": Object {
          "description": "Tag must be paired.",
          "id": "tag-pair",
          "link": "https://github.com/thedaviddias/HTMLHint/wiki/tag-pair",
        },
        "type": "error",
      },
      Object {
        "col": 3,
        "evidence": "  <div class\\"foo\\">",
        "line": 11,
        "message": "Special characters must be escaped : [ < ].",
        "raw": "
  <div class\\"foo\\">

          ",
        "rule": Object {
          "description": "Special characters must be escaped.",
          "id": "spec-char-escape",
          "link": "https://github.com/thedaviddias/HTMLHint/wiki/spec-char-escape",
        },
        "type": "error",
      },
      Object {
        "col": 18,
        "evidence": "  <div class\\"foo\\">",
        "line": 11,
        "message": "Special characters must be escaped : [ > ].",
        "raw": "
  <div class\\"foo\\">

          ",
        "rule": Object {
          "description": "Special characters must be escaped.",
          "id": "spec-char-escape",
          "link": "https://github.com/thedaviddias/HTMLHint/wiki/spec-char-escape",
        },
        "type": "error",
      },
      Object {
        "col": 11,
        "evidence": "          </div>",
        "line": 13,
        "message": "Tag must be paired, no start tag: [ </div> ]",
        "raw": "</div>",
        "rule": Object {
          "description": "Tag must be paired.",
          "id": "tag-pair",
          "link": "https://github.com/thedaviddias/HTMLHint/wiki/tag-pair",
        },
        "type": "error",
      },
      Object {
        "col": 9,
        "evidence": "        </div>",
        "line": 14,
        "message": "Tag must be paired, no start tag: [ </div> ]",
        "raw": "</div>",
        "rule": Object {
          "description": "Tag must be paired.",
          "id": "tag-pair",
          "link": "https://github.com/thedaviddias/HTMLHint/wiki/tag-pair",
        },
        "type": "error",
      },
      Object {
        "col": 7,
        "evidence": "      </hello>",
        "line": 15,
        "message": "Tag must be paired, no start tag: [ </hello> ]",
        "raw": "</hello>",
        "rule": Object {
          "description": "Tag must be paired.",
          "id": "tag-pair",
          "link": "https://github.com/thedaviddias/HTMLHint/wiki/tag-pair",
        },
        "type": "error",
      },
      Object {
        "col": 5,
        "evidence": "    </test>",
        "line": 16,
        "message": "Tag must be paired, no start tag: [ </test> ]",
        "raw": "</test>",
        "rule": Object {
          "description": "Tag must be paired.",
          "id": "tag-pair",
          "link": "https://github.com/thedaviddias/HTMLHint/wiki/tag-pair",
        },
        "type": "error",
      },
      Object {
        "col": 3,
        "evidence": "  </div>",
        "line": 17,
        "message": "Tag must be paired, no start tag: [ </div> ]",
        "raw": "</div>",
        "rule": Object {
          "description": "Tag must be paired.",
          "id": "tag-pair",
          "link": "https://github.com/thedaviddias/HTMLHint/wiki/tag-pair",
        },
        "type": "error",
      },
      Object {
        "col": 15,
        "evidence": "          <div class='' what='' something=''></div>",
        "line": 21,
        "message": "The value of attribute [ class ] must be in double quotes.",
        "raw": " class=''",
        "rule": Object {
          "description": "Attribute values must be in double quotes.",
          "id": "attr-value-double-quotes",
          "link": "https://github.com/thedaviddias/HTMLHint/wiki/attr-value-double-quotes",
        },
        "type": "error",
      },
      Object {
        "col": 24,
        "evidence": "          <div class='' what='' something=''></div>",
        "line": 21,
        "message": "The value of attribute [ what ] must be in double quotes.",
        "raw": " what=''",
        "rule": Object {
          "description": "Attribute values must be in double quotes.",
          "id": "attr-value-double-quotes",
          "link": "https://github.com/thedaviddias/HTMLHint/wiki/attr-value-double-quotes",
        },
        "type": "error",
      },
      Object {
        "col": 32,
        "evidence": "          <div class='' what='' something=''></div>",
        "line": 21,
        "message": "The value of attribute [ something ] must be in double quotes.",
        "raw": " something=''",
        "rule": Object {
          "description": "Attribute values must be in double quotes.",
          "id": "attr-value-double-quotes",
          "link": "https://github.com/thedaviddias/HTMLHint/wiki/attr-value-double-quotes",
        },
        "type": "error",
      },
      Object {
        "col": 3,
        "evidence": "  </div>",
        "line": 25,
        "message": "Tag must be paired, no start tag: [ </div> ]",
        "raw": "</div>",
        "rule": Object {
          "description": "Tag must be paired.",
          "id": "tag-pair",
          "link": "https://github.com/thedaviddias/HTMLHint/wiki/tag-pair",
        },
        "type": "error",
      },
      Object {
        "col": 1,
        "evidence": "</bad>",
        "line": 26,
        "message": "Tag must be paired, no start tag: [ </bad> ]",
        "raw": "</bad>",
        "rule": Object {
          "description": "Tag must be paired.",
          "id": "tag-pair",
          "link": "https://github.com/thedaviddias/HTMLHint/wiki/tag-pair",
        },
        "type": "error",
      },
      Object {
        "col": 1,
        "evidence": "</bad>",
        "line": 27,
        "message": "Tag must be paired, no start tag: [ </bad> ]",
        "raw": "</bad>",
        "rule": Object {
          "description": "Tag must be paired.",
          "id": "tag-pair",
          "link": "https://github.com/thedaviddias/HTMLHint/wiki/tag-pair",
        },
        "type": "error",
      },
    ],
    "time": 99,
  },
]
`;
